src/som2text.rs

```diff
@@ -258,6 +258,53 @@ mod optionals {
     }
 }
 
+mod bitfields {
+    use super::*;
+    use crate::som::bitfields::*;
+
+    impl Display for SOMBitfieldMemberType {
+        fn fmt(&self, f: &mut Formatter) -> FmtResult {
+            let (d, b) = match self {
+                SOMBitfieldMemberType::U8(item) => (item.value().to_string(), item.bit_len()),
+                SOMBitfieldMemberType::U16(item) => (item.value().to_string(), item.bit_len()),
+                SOMBitfieldMemberType::U32(item) => (item.value().to_string(), item.bit_len()),
+                SOMBitfieldMemberType::U64(item) => (item.value().to_string(), item.bit_len()),
+            };
+
+            write!(f, "{}<{}>", d, b)
+        }
+    }
+
+    impl Display for SOMBitfieldType {
+        fn fmt(&self, f: &mut Formatter) -> FmtResult {
+            let mut d = match self.meta() {
+                Some(meta) => meta.to_str(),
+                None => String::from("bitfield"),
+            };
+
+            let mut dd = String::from("");
+            for i in 0..self.len() {
+                if let Some(member) = self.get(i) {
+                    let ddd = format!("{}", member);
+                    dd = format!("{}\n\t{},", dd, ddd.replace('\n', "\n\t"));
+                }
+            }
```

Comment:
adding to a string can be quite expensive, the recommended way is:
1.pre-allocate a string
2.push into that string
3.for formatted strings, use the `write!` macro
```
use std::fmt::Write; // Import Write trait for using write! on String
let mut result = String::with_capacity(100); // Pre-allocate space for 100 characters
result.push_str("Hello");
result.push(' ');
result.push_str("world!");
result.push('!');
write!(&mut result, "{}", some_variable).unwrap();
```

Reply:
The code adds nested (and possible complex) member strings to a parent type string which need to be intended accordingly, thus requiring a replace on the nested member strings:

` 
let ddd = format!("{}", member);
dd = format!("{}\n\t{},", dd, ddd.replace('\n', "\n\t"));
`

The alternative would be using a `pretty_print(intent)` API for building up the type-strings instead using `Display` within the whole som2text module. This would be a major change for another PR.

As for using `String::with_capacity(estimated);` the trade off would be that many resulting strings may be quite smaller than `estimated` thus raising unnecessary pre-allocations, while others would be quite bigger due to the nested type layout and raising re-allocations,- versus the current building up the string as-it-comes.